### PR TITLE
Fix mysqldb1

### PR DESCRIPTION
**Pull request by daveh86**
This fixes the missing mysql package dependency on mysqldb1, and of course adds mysql

### CODE REVIEW

**Comment by daveh86** [comment]
I've tested this locally on my SLES12 setup and this works on that system to build both mysql and add py-mysqldb1

File: var/spack/repos/builtin/packages/mysql/package.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class Mysql(CMakePackage):
+    homepage = "http://dev.mysql.com"
+    url      = "https://dev.mysql.com/get/Downloads/MySQL-8.0/mysql-8.0.11.tar.gz"
+
+    version('8.0.11', '38d5a5c1a1eeed1129fec3a999aa5efd')
+    version('5.7.22', '269935a8b72dcba2c774d8d63a8bd1dd')
+
+    depends_on('boost@1.66.0', when='@8.0.11')
```
Comment (healther) [inline var/spack/repos/builtin/packages/mysql/package.py]: does it only work with these specific boost versions?

**Comment by adamjstewart** [comment]
I need to install mysql and postgresql for my research, so I'm going to merge this and test it out. If I find any problems, I'll just submit an additional PR.

File: var/spack/repos/builtin/packages/mysql/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/mysql/package.py]: Yes, each version of `mysql` requires a specific version of `boost` at build time. See `BOOST_PACKAGE_NAME` in `cmake/boost.cmake` in the downloaded tarball.

> The Boost C++ libraries are required to build MySQL (but not to use it). MySQL compilation requires a particular Boost version. Typically, that is the current Boost version, but if a specific MySQL source distribution requires a different version, the configuration process will stop with a message indicating the Boost version that it requires.

https://dev.mysql.com/doc/refman/8.0/en/source-installation.html

**Comment by chissg** [comment]
@adamjstewart I am preparing a PR for this which provides many more versions (including from before Boost was required), treats `boost` as the build-only dependency it is, and ensures that Boost is picked up from the correct place. It also exposed a deficiency in the `boost` recipe which will be addressed in a separate PR.

**Comment by chissg** [comment]
@adamjstewart Just saw PR #10904, will comment over there.

**Pull request merged**